### PR TITLE
app-editors/gedit,  app-editors/gedit-plugins: Version bump to 45.0, gui-libs/tepl: Version bump to 6.6.0, gui-libs/libgedit-gtksourceview: New ebuild 

### PR DESCRIPTION
**Pull request by flint2**
gui-libs/libgedit-gtksourceview is a new dependency for gui-libs/tepl

### CODE REVIEW

File: app-editors/gedit/gedit-45.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{9..11} )
```
Comment (thesamesam) [inline app-editors/gedit/gedit-45.0.ebuild:5]: 9 doesn't do anything anymore, drop it when you see it

**Comment by mattst88** [comment]
The QA report says

````
libgedit-gtksourceview
--
299.0.3 
  | TrailingEmptyLine | ebuild has trailing blank line(s)
  | UnusedInherits | unused eclass: virtualx
````

**Comment by mattst88** [comment]
Shouldn't we include gedit-plugins in this PR?

**Comment by mattst88** [comment]
All libgedit-gtksourceview's tests fail because they need an X display.

**Comment by mattst88** [comment]
```
LIBGEDIT_GTKSOURCEVIEW_PUBLIC_DEPS = [
  dependency('gio-2.0', version: '>= 2.74'),
  dependency('gtk+-3.0', version: '>= 3.20'),
]
```

But the ebuild contains

```
	>=dev-libs/glib-2.56:2
	>=x11-libs/gtk+-3.22:3
```

I know you can do better than this. :)

**Comment by mattst88** [comment]
```
./usr/lib64/pkgconfig/libgedit-gtksourceview-300.pc
./usr/lib64/girepository-1.0/GtkSource-300.typelib
./usr/lib64/libgedit-gtksourceview-300.so.0
```

Why `SLOT=4`? If anything, shouldn't it be 0 or 300?

**Comment by flint2** [comment]
> All libgedit-gtksourceview's tests fail because they need an X display.

Solved in https://github.com/gentoo/gentoo/pull/31620/commits/1ecd5c0398db5623a78cb2b1b24f7520a3a301cb

**Comment by flint2** [comment]
> I don't think anything requires `tepl[introspection]`? Can we just drop IUSE=introspection on tepl and libgedit-gtksourceview and leave it disabled?

Done in https://github.com/gentoo/gentoo/pull/31620/commits/cf5dff219b661ab67e0f73599ad13253009acf3f and https://github.com/gentoo/gentoo/pull/31620/commits/1ecd5c0398db5623a78cb2b1b24f7520a3a301cb

**Comment by flint2** [comment]
> ```
> LIBGEDIT_GTKSOURCEVIEW_PUBLIC_DEPS = [
>   dependency('gio-2.0', version: '>= 2.74'),
>   dependency('gtk+-3.0', version: '>= 3.20'),
> ]
> ```
> 
> But the ebuild contains
> 
> ```
> 	>=dev-libs/glib-2.56:2
> 	>=x11-libs/gtk+-3.22:3
> ```
> 
> I know you can do better than this. :)

I 'm a pkgbump junkie, new packages are a different level :-)

**Comment by flint2** [comment]
> ```
> ./usr/lib64/pkgconfig/libgedit-gtksourceview-300.pc
> ./usr/lib64/girepository-1.0/GtkSource-300.typelib
> ./usr/lib64/libgedit-gtksourceview-300.so.0
> ```
> 
> Why `SLOT=4`? If anything, shouldn't it be 0 or 300?

Corrected in https://github.com/gentoo/gentoo/pull/31620/commits/1ecd5c0398db5623a78cb2b1b24f7520a3a301cb , was because is a copy from other ebuild and forgot to check it.

**Comment by mattst88** [comment]
I've made these changes locally:

```diff

diff --git a/gui-libs/libgedit-gtksourceview/libgedit-gtksourceview-299.0.3.ebuild b/gui-libs/libgedit-gtksourceview/libgedit-gtksourceview-299.0.3.ebuild
index 593eefe2bebf..6617bc8d1fbf 100644
--- a/gui-libs/libgedit-gtksourceview/libgedit-gtksourceview-299.0.3.ebuild
+++ b/gui-libs/libgedit-gtksourceview/libgedit-gtksourceview-299.0.3.ebuild
@@ -10,16 +10,17 @@ HOMEPAGE="https://github.com/gedit-technology/libgedit-gtksourceview"
 SRC_URI="https://gedit-technology.net/tarballs/libgedit-gtksourceview/${P}.tar.xz"
 
 LICENSE="LGPL-2.1+"
-SLOT="0"
+SLOT="300"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
 
 IUSE="gtk-doc"
 
-KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
-
 RDEPEND="
 	>=dev-libs/glib-2.74:2
 	>=x11-libs/gtk+-3.20:3
 	>=dev-libs/libxml2-2.6:2
+
+	dev-libs/gobject-introspection:=
 "
 DEPEND="${RDEPEND}"
 BDEPEND="
@@ -34,6 +35,7 @@ BDEPEND="
 
 src_configure() {
 	local emesonargs=(
+		-Dgobject_introspection=true
 		$(meson_use gtk-doc gtk_doc)
 	)
 	meson_src_configure
-- 
```

```diff
diff --git a/gui-libs/tepl/tepl-6.6.0.ebuild b/gui-libs/tepl/tepl-6.6.0.ebuild
index 4a28860c4a82..9131540148af 100644
--- a/gui-libs/tepl/tepl-6.6.0.ebuild
+++ b/gui-libs/tepl/tepl-6.6.0.ebuild
@@ -17,12 +17,14 @@ RESTRICT="!test? ( test )"
 RDEPEND="
 	>=dev-libs/glib-2.62:2
 	>=x11-libs/gtk+-3.22:3
-	>=x11-libs/gtksourceview-4.0:4
+	>=gui-libs/libgedit-gtksourceview-299.0.1:300
 	>=gui-libs/amtk-5.0:5=
-	>=gui-libs/libgedit-gtksourceview-299:4
 	dev-libs/icu:=
 	gnome-base/gsettings-desktop-schemas
+
+	dev-libs/gobject-introspection:=
 "
+DEPEND="${RDEPEND}"
 BDEPEND="
 	dev-util/glib-utils
 	gtk-doc? (
@@ -35,6 +37,7 @@ BDEPEND="
 
 src_configure() {
 	local emesonargs=(
+		-Dgobject_introspection=true
 		$(meson_use gtk-doc gtk_doc)
 	)
 	meson_src_configure
-- 
```

```diff
diff --git a/app-editors/gedit/gedit-45.0.ebuild b/app-editors/gedit/gedit-45.0.ebuild
index 580e018942aa..858bf177b846 100644
--- a/app-editors/gedit/gedit-45.0.ebuild
+++ b/app-editors/gedit/gedit-45.0.ebuild
@@ -11,17 +11,16 @@ HOMEPAGE="https://wiki.gnome.org/Apps/Gedit"
 
 LICENSE="GPL-2+ CC-BY-SA-3.0"
 SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
 
 IUSE="+python gtk-doc"
 REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
 
-KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
-
 DEPEND="
 	>=dev-libs/glib-2.70:2
 	>=x11-libs/gtk+-3.22.0:3[introspection]
 	>=gui-libs/amtk-5.6:=
-	>=gui-libs/tepl-6.5:=
+	>=gui-libs/tepl-6.5.1:=
 	>=dev-libs/libpeas-1.14.1[gtk]
 	>=dev-libs/gobject-introspection-1.54:=
 	>=app-text/gspell-0.2.5:0=
@@ -34,7 +33,7 @@ DEPEND="
 		')
 	)
 
-	>=x11-libs/gtksourceview-4.0.2:4[introspection,vala]
+	gui-libs/libgedit-gtksourceview:300
 "
 RDEPEND="${DEPEND}
 	x11-themes/adwaita-icon-theme
@@ -43,7 +42,6 @@ RDEPEND="${DEPEND}
 "
 BDEPEND="
 	$(vala_depend)
-	app-text/docbook-xml-dtd:4.1.2
 	dev-util/glib-utils
 	gtk-doc? ( >=dev-util/gtk-doc-1 )
 	dev-util/itstool
-- 
```

```diff
diff --git a/app-editors/gedit-plugins/gedit-plugins-45.0.ebuild b/app-editors/gedit-plugins/gedit-plugins-45.0.ebuild
index d7b036c46cf7..10479b810a46 100644
--- a/app-editors/gedit-plugins/gedit-plugins-45.0.ebuild
+++ b/app-editors/gedit-plugins/gedit-plugins-45.0.ebuild
@@ -27,7 +27,7 @@ RDEPEND="
 	>=app-editors/gedit-45.0
 	>=dev-libs/glib-2.32:2
 	>=x11-libs/gtk+-3.9:3
-	>=x11-libs/gtksourceview-4.0.2:4
+	gui-libs/libgedit-gtksourceview:300
 	python? (
 		${PYTHON_DEPS}
 		$(python_gen_cond_dep '
@@ -38,7 +38,6 @@ RDEPEND="
 			dev-python/pygobject:3[cairo,${PYTHON_USEDEP}]
 		')
 		>=x11-libs/gtk+-3.9:3[introspection]
-		>=x11-libs/gtksourceview-4.0.2:4[introspection]
 		x11-libs/pango[introspection]
 		x11-libs/gdk-pixbuf:2[introspection]
 	)
-- 
```

The main problems were

- Turns out we need introspection enabled in tepl for gedit
- We need to depend on libgedit-gtksourceview instead of gtksourceview

**Pull request closed without merging**